qiime2/sdk/plugin_manager.py

```diff
@@ -142,34 +144,75 @@ def get_semantic_types(self):
 
         return types
 
+    class FormatFilters(enum.Flag):
+        EXPORTABLE = enum.auto()
+        IMPORTABLE = enum.auto()
+
     # TODO: Should plugin loading be transactional? i.e. if there's
     # something wrong, the entire plugin fails to load any piece, like a
     # databases rollback/commit
 
-    def get_formats(self, *, include_all=False, importable=False,
-                    exportable=False, canonical_format=False):
-        if include_all is True and canonical_format is True or include_all \
-                is True and importable is True or include_all is True and \
-                exportable is True:
-            raise ValueError("If all formats are requested, other formats "
-                             "cannot be included as a result.")
-
-        elif include_all is True:
-            return self.formats
+    def get_formats(self, *, filter=None, semantic_type=None):
```

Comment:
This method will need a [numpydoc-compatible](https://numpydoc.readthedocs.io/en/latest/format.html) docstring.